packages/elements/list/src/insertFragmentList.ts

```diff
@@ -49,35 +73,134 @@ export const insertFragmentList = (editor: PlateEditor) => {
   return (fragment: TDescendant[]) => {
     const liEntry = findNode(editor, {
-      match: { type: li.type },
+      match: { type: listItemType },
       mode: 'lowest',
     });
+    // not inserting into a list item, delegate to other plugins
+    if (!liEntry) {
+      return insertFragment(
+        isListRoot(fragment[0]) ? [{ text: '' }, ...fragment] : fragment
+      );
+    }
 
-    if (liEntry) {
-      const [, liPath] = liEntry;
+    // delete selection (if necessary) so that it can check if needs to insert into an empty block
+    Transforms.insertFragment(editor, [{ text: '' }]);
 
-      // FIXME: fork insertFragment for edge cases
-      return Transforms.insertNodes(
-        editor,
-        fragment.flatMap((node) => trimList(node)),
-        { at: Path.next(liPath), select: true }
+    const licEntry = findNode(editor, {
+      match: { type: listItemContentType },
+      mode: 'lowest',
+    });
+    if (!licEntry) {
+      return insertFragment(
+        isListRoot(fragment[0]) ? [{ text: '' }, ...fragment] : fragment
       );
     }
 
-    const filtered: TDescendant[] = isListRoot(fragment[0])
-      ? [{ text: '' }, ...fragment]
-      : fragment;
+    const [, liPath] = liEntry!;
+    const [licNode] = licEntry!;
+    const isEmptyNode = isCollapsed(editor.selection) && !Node.string(licNode);
+    const [first, ...rest] = fragment
+      .flatMap(trimList)
+      .map(wrapNodeIntoListItem);
+
+    let nodes;
+    let firstText;
+    if (isListRoot(fragment[0])) {
+      if (isSingleLic(fragment)) {
+        firstText = first;
+        nodes = rest;
+      } else if (isEmptyNode) {
+        // FIXME: is there a more direct way to set this?
+        const li = Node.get(editor, liPath) as Element;
+        const [, ...currentSublists] = li.children;
+        const [newLic, ...newSublists] = first.children;
+        // // it will set the LIC to the value of the item to be inserted,
+        // // will keep the existing sublists (if any)
+        // // and will insert the new sublists
+        if (newSublists?.length && isListRoot(newSublists[0])) {
+          if (currentSublists?.length && isListRoot(currentSublists[0])) {
+            (currentSublists[0] as Element).children = [
+              ...(currentSublists[0] as Element).children,
+              ...(newSublists[0] as Element).children,
+            ];
+
+            li.children = [newLic, ...currentSublists];
+          } else {
+            li.children = [newLic, ...newSublists];
+          }
+        } else {
+          li.children = [newLic, ...currentSublists];
+        }
 
-    return insertFragment(filtered);
+        firstText = { text: '' };
+        nodes = rest;
+      } else {
+        firstText = { text: '' };
+        nodes = [first, ...rest];
+      }
+    } else {
+      firstText = first;
+      nodes = rest;
+    }
+    Transforms.insertFragment(editor, [firstText]); // insert text if needed
+
+    return Transforms.insertNodes(editor, nodes, {
```

Comment:
The rest of the LIs will be inserted as siblings of the selected LI, just as it works now.